src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -23,14 +26,17 @@ class remove_virtual_functionst
 {
 public:
   remove_virtual_functionst(
-    const symbol_table_baset &_symbol_table,
+    symbol_table_baset &_symbol_table,
     const class_hierarchyt &_class_hierarchy);
 
   void operator()(goto_functionst &goto_functions);
 
-  bool remove_virtual_functions(goto_programt &goto_program);
+  bool remove_virtual_functions(
+    irep_idt function_identifier,
```

Comment:
```suggestion
    const irep_idt &function_id,
```

const & (several occurrences)

Also, I think we settled on calling this `function_id` throughout the codebase.